[InputGen] introduce variable types

manuelcandales opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #116138
* #116137
* #116136
* #116135
* #115698
* #115697
* #115696
* #115695
* #115694
* #115693
* #115692
* #115691
* #115605
* #115604
* #115603
* __->__ #115602
* #115601

This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

manuelcandales committed:
[InputGen] introduce variable types

This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

larryliu0820 commented on torch/testing/_internal/inputgen/variable/type.py:
```diff
@@ -0,0 +1,70 @@
+    # torch.float16,
+    torch.float32,
+    torch.float64,
+    # torch.complex32,
+    # torch.complex64,
+    # torch.complex128,
+    # torch.bfloat16,
```
why commented out?

manuelcandales replied:
Because I want to support these dtypes in the short term. But I haven't verified my specifications with them yet.

larryliu0820 commented on torch/testing/_internal/inputgen/variable/type.py:
```diff
@@ -0,0 +1,70 @@
+]
+
+
+class VariableType(Enum):
```
We should have some docstring that sits nearby the code

larryliu0820 commented on torch/testing/_internal/inputgen/variable/type.py:
```diff
@@ -0,0 +1,70 @@
+        return
+    if vtype in [bool, int, float]:
+        if type(v) not in [bool, int, float]:
+            raise TypeError("Variable type mismatch")
```
Do we really need to raise here? Or we can return true/false.

manuelcandales replied:
Yes, we can return true/false, but then I will have to repeat that line `raise TypeError("Variable type mismatch")` every time.

manuelcandales replied:
Ok, updated the code. Now it returns boolean.

larryliu0820 commented on torch/testing/_internal/inputgen/variable/type.py:
```diff
@@ -0,0 +1,70 @@
+    return int(v) == v
+
+
+def convert_to_vtype(vtype, v) -> None:
```
This return hint doesn't seem right.

manuelcandales replied:
Fixed.

larryliu0820 commented on torch/testing/_internal/inputgen/variable/type.py:
```diff
@@ -0,0 +1,70 @@
+            raise TypeError("Variable type mismatch")
+
+
+def is_integer(v):
```
I think we should adapt to the coding convention of having strict type annotation, given we are pretty sensitive on types.

manuelcandales replied:
Updated!
Yes, I agree. That's my goal. I am sure I still have a bunch of spots where I missed them though.

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]

manuelcandales committed:
Update on "[InputGen] introduce variable types"


This is the first in a stack of diffs aimed at landing InputGen, a test generation engine for pytorch ops

This diff introduces the most basic variable types that are generated. These are the building blocks for generating more complex structures. These variable types are:
- Bool (True or False)
- Int (any integral value)
- Float (any floating point value)
- String (any string)
- ScalarDtype (currently this can take only 3 values: bool, int or float)
- TensorDtype
- Tuple (any tuple of values)

Differential Revision: [D52047188](https://our.internmc.facebook.com/intern/diff/D52047188/)

[ghstack-poisoned]
